File: backend/src/main/java/com/paike/scheduler/entity/SystemAuditLog.java
package com.paike.scheduler.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@TableName("system_audit_log")
public class SystemAuditLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long operatorId;

    private String operatorName;

    private String actionType;

    private String targetType;

    private Long targetId;

    private Long semesterId;

    private Long planId;

    private Integer success;

    private String beforeSummary;

    private String afterSummary;

    private String errorCode;

    private String errorMessage;

    @TableField("created_at")
    private LocalDateTime createdAt;
}

File: frontend/src/api/systemAuditLog.ts
import request from '../utils/request'
import type { ApiResponse, PageResult } from './types'

export interface SystemAuditLog {
  id: number
  operatorId: number | null
  operatorName: string | null
  actionType: string
  targetType: string
  targetId: number | null
  semesterId: number | null
  planId: number | null
  success: number
  beforeSummary: string | null
  afterSummary: string | null
  errorCode: string | null
  errorMessage: string | null
  createdAt: string
}

export interface SystemAuditLogQuery {
  actionType?: string
  semesterId?: number
  planId?: number
  success?: boolean
  page?: number
  size?: number
}

export function getSystemAuditLogList(params: SystemAuditLogQuery) {
  return request.get<ApiResponse<PageResult<SystemAuditLog>>>('/v6/audit-logs', { params }).then((r) => {
    if (!r.data) throw new Error('响应数据为空')
    return r.data.data
  })
}

export function getSystemAuditLogById(id: number) {
  return request.get<ApiResponse<SystemAuditLog>>(`/v6/audit-logs/${id}`).then((r) => {
    if (!r.data) throw new Error('响应数据为空')
    return r.data.data
  })
}
